sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1644,12 +1755,29 @@ def task_worker(self):
                             self.log_notice("{} waiting for host tx ready...".format(lport))
                             continue
 
-                        # D.1.3 Software Configuration and Initialization
-                        api.set_datapath_init(host_lanes_mask)
-                        dpInitDuration = self.get_cmis_dp_init_duration_secs(api)
-                        self.log_notice("{}: DpInit duration {} secs".format(lport, dpInitDuration))
-                        self.port_dict[lport]['cmis_expired'] = now + datetime.timedelta(seconds=dpInitDuration)
-                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_TXON
+                        optics_si_dict = {}
+                        if not self.port_dict[lport]['optics_si_validated'] and g_optics_si_dict:
```

Comment:
@AnoopKamath not sure the use of adding this check for 'optics_si_validated' because we have reached  at this point from DP deinit so at this point we have to apply the SI settings(once we know platform needs it) before DP init

Reply:
@prgeor : optics_si_validated flag is used to avoid calling or skip SI settings api again in point 2 below.
1. Validate config -> set SI settings without invoking  set_datapath_init()
2. Validate config -> invoke set_datapath_init() without set SI settings.

Reply:
Removed optics_si_validated and calling stage_custom_si_settings() only once